crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -337,13 +332,13 @@ fn match_settled_prices(
     solver_prices: HashMap<H160, U256>,
 ) -> Result<HashMap<H160, U256>> {
     let mut prices = HashMap::new();
-    let executed_tokens = executed_limit_orders.iter().flat_map(|order| {
-        if order.order.is_liquidity_order {
-            vec![order.order.sell_token]
-        } else {
-            vec![order.order.buy_token, order.order.sell_token]
-        }
-    });
+    let executed_tokens = executed_limit_orders
+        .iter()
+        .flat_map(|order| match order.order.id {
+            LimitOrderId::Market(_) => vec![order.order.buy_token, order.order.sell_token],
+            LimitOrderId::Limit(_) => vec![],
+            LimitOrderId::Liquidity(_) => vec![],
+        });
     for token in executed_tokens {
```

Comment:
Second bug fixed. This is a follow up to https://github.com/cowprotocol/services/pull/878 where I missed to update this line. In https://github.com/cowprotocol/services/pull/878 we decided to decouple prices for limit and liquidity orders from UCP.

Reply:
I think we still need to keep around the uniform clearing prices for tokens traded by `Limit` orders.

Reply:
Why do you think that?
All changes in https://github.com/cowprotocol/services/pull/878 were made around the idea of both liquidity and limit orders having custom prices therefore being removed from the UCP.

Reply:
The [code](https://github.com/cowprotocol/services/blob/main/crates/solver/src/settlement/settlement_encoder.rs#L303-L310) that later computes the price adjusted for the `surplus_fee` in the `SettlementEncoder` still needs to know the uniform clearing price of the tokens traded in the limit order.
If we don't collect the traded tokens of the limit order here we might end up dropping their uniform clearing prices in `match_settled_prices()`.

Reply:
Also it would be nice to know the original UCP for the limit order tokens in the `settle` transaction to prove that our custom prices still respect our promise of UCP (they just got adjusted with different `surplus_fee` amounts).

Reply:
Yes, we definitely need to keep the UCPs in for limit orders.

As an example - a settlement that trades exactly 1 limit order should have 4 prices:
- UCP sell token price
- UCP buy token price
- limit order sell token price
- limit order buy token price

This is a soft requirement so that we can verify UCP after the fact.

Reply:
Indeed. Added tokens for limit orders.